src/porepy/numerics/ad/_ad_utils.py

```diff
@@ -626,7 +626,7 @@ def parse(self, mdg: pp.MixedDimensionalGrid) -> sps.spmatrix:
             mat.append(local_mat)
 
         if all([isinstance(m, np.ndarray) for m in mat]):
-            # TODO: EK is almost sure this never happens, but leave this check for now.
+            # EK is almost sure this never happens, but leave this check for now.
```

Comment:
Should we also remove the check?

Reply:
If you don't mind, I prefer to keep it there as long as we occasionally work on the Ad parsing.  I think I was actually saved by that if some weeks ago.